clients/admin-ui/src/features/dataset/ApproveClassification.tsx

```diff
@@ -43,25 +43,40 @@ const ApproveClassification = () => {
 
     const updatedDataset = getUpdatedDatasetFromClassifyDataset(
       dataset,
-      classifyDataset
+      classifyDataset,
+      classifyCollection?.name
     );
     try {
       const updateResult = await updateDataset(updatedDataset);
       if (isErrorResult(updateResult)) {
         toast(errorToastParams(getErrorMessage(updateResult.error)));
         return;
       }
-
-      const statusResult = await updateClassifyInstance({
-        dataset_fides_key: dataset.fides_key,
-        status: ClassificationStatus.REVIEWED,
+      toast(successToastParams("Collection classified and approved"));
+      let uncategorizedCount = 0;
+      updatedDataset.collections.forEach((updatedCollection) => {
+        updatedCollection.fields.forEach((updatedField) => {
+          if (
+            !updatedField.data_categories ||
+            updatedField.data_categories.length === 0
+          ) {
+            uncategorizedCount += 1;
+          }
+        });
       });
-      if (isErrorResult(statusResult)) {
-        toast(errorToastParams(getErrorMessage(statusResult.error)));
-        return;
-      }
+      if (uncategorizedCount === 0) {
```

Comment:
ah okay it took me a while to figure out what was happening here. might be helpful to add a comment along the lines of `Only update the dataset as classified when all fields have been categorized` ?

Reply:
comments added 👍🏽